### PR TITLE
Server4

### PR DESCRIPTION
**Pull request by zhangshyue**
Add option to upvote the comments. 
Screenshot 2020-07-06 at 8 36 36 AM

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/UpvoteServlet.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        long id = Long.parseLong(request.getParameter("id"));
+        Key key = KeyFactory.createKey("Entry", id);
+        // Update the number of upvotes of the comment that is been clicked.
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        try {
+            Entity commentEntity = datastore.get(key);
+            int upvote = Integer.parseInt(commentEntity.getProperty("upvote").toString()) + 1;
+            commentEntity.setProperty("upvote", upvote);
+            datastore.put(commentEntity);
+        }catch (EntityNotFoundException e) {
```
Comment (MorganAw) [inline portfolio/src/main/java/com/google/sps/servlets/UpvoteServlet.java]: styling nit: add a space
`} catch (EntityNotFoundException ...`

File: portfolio/src/main/webapp/comment.js [hunk 2]
```diff
@@ -29,19 +29,46 @@ function getContentFunctions() {
...
+    upvoteElement.appendChild(upvoteButtonElement);
+    cardBodyElement.appendChild(upvoteElement);
+    divElement.appendChild(cardBodyElement);
+
     return divElement;
+}
+
+/** Update the number of upvotes of the comment that is been upvoted. */
+function updateUpvote(text, num) {
+    const params = new URLSearchParams();
```
Comment (MorganAw) [inline portfolio/src/main/webapp/comment.js:68]: nit: Seems like this works but I think a regular JSON object would work here. URLSearchParams is generally usually used in relation to URL manipulation.

File: portfolio/src/main/webapp/comment.js [hunk 3]
```diff
@@ -29,19 +29,46 @@ function getContentFunctions() {
...
+    params.append('id', text.id);
+    fetch('/update-upvote', {method: 'POST', body: params});
+    const currentText = document.getElementsByClassName(num)[0].innerText;
```
Comment (MorganAw) [inline portfolio/src/main/webapp/comment.js]: Could wrap the rest of this in a `.then` in case the update were to fail for some reason?

**Review by MorganAw — APPROVED** [review]

File: portfolio/src/main/webapp/comment.js (hunk 2 above)
Comment (zhangshyue) [reply portfolio/src/main/webapp/comment.js:68]: That's true, but I'm not sure how to create a regular JSON object. I tried const params = JSON.stringify({'id': text.id}); and const params = {'id': text.id}; but it all gives me Internet server error.

**Pull request merged**